src/libs/ajax/User.ts

```diff
@@ -74,17 +94,44 @@ export interface SetTerraUserProfileRequest {
   interestInTerra?: string;
 }
 
-export const makeSetUserProfileRequest = (terraUserProfile: TerraUserProfile): SetTerraUserProfileRequest => {
+export const generateAPIBodyForCreateUserProfile = (
+  request: CreateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
+  return generateAPIBodyForUpdateUserProfile(request);
+};
+
+export const generateAPIBodyForUpdateUserProfile = (
+  request: UpdateTerraUserProfileRequest
+): OrchestrationUpsertTerraUserProfileRequest => {
   return {
-    firstName: terraUserProfile.firstName ?? 'N/A',
-    lastName: terraUserProfile.lastName ?? 'N/A',
-    title: terraUserProfile.title ?? 'N/A',
-    institute: terraUserProfile.institute ?? 'N/A',
-    programLocationCity: terraUserProfile.programLocationCity ?? 'N/A',
-    programLocationState: terraUserProfile.programLocationState ?? 'N/A',
-    programLocationCountry: terraUserProfile.programLocationCountry ?? 'N/A',
-    department: terraUserProfile.department,
-    interestInTerra: terraUserProfile.interestInTerra,
+    // TODO: this is the logic to look at when writing the tests
+    // first name and last name are REQUIRED for this request, and they are populated from the oidc token claims,
+    // so they should not be undefined
+    firstName: request.firstName!,
+    lastName: request.lastName!,
```

Comment:
`firstName` and `lastName` are typed as `string` in `UpdateTerraUserProfileRequest`. Since their type does not include null/undefined, this non-null assertion should be unnecessary.

```suggestion
    firstName: request.firstName,
    lastName: request.lastName,
```